### ISSUE TITLE
`external-name` intermittently changes

### ISSUE THREAD

**Issue by fernandezcuesta** [crossplane-contrib/provider-upjet-aws#610]
### What happened?

For some managed objects (such as `subnets.ec2.aws.upbound.io`) the external name gets updated from the explicitly set (e.g. `private-subnet-0-us-east-2-main`) to the actual subnet-id (as per AWS):

![image](https://user-images.githubusercontent.com/7312236/225913933-ea5884cd-cab7-417b-8d4f-feb338ad61c4.png)

### How can we reproduce it?

Composition including some AWS resource (e.g. `kind: Subnet`) and patch setting the `external-name` annotation such as:

```
    - name: private-subnet-0
      base:
        apiVersion: ec2.aws.upbound.io/v1beta1
        kind: Subnet
        metadata:
          labels: &labels
            type: subnet
            nx-subnet-visibility: private
            nx-component: eks
        spec:
          forProvider:
            mapPublicIpOnLaunch: false
            tags:
              <<: *labels
              kubernetes.io/role/internal-elb: "1"
      patches:
        - *commonParametersPatch
        [... REDACTED ...]
        - &setName
          type: CombineFromComposite
          toFieldPath: metadata.name
          combine:
            variables:
            - fromFieldPath: spec.resourceConfig.region
            - fromFieldPath: spec.parameters.clusterName
            strategy: string
            string:
              fmt: "private-subnet-0-%s-%s"
        - <<: *setName
          toFieldPath: metadata.annotations["crossplane.io/external-name"]
        - <<: *setName
          toFieldPath: spec.forProvider.tags.Name
```

### What environment did it happen in?

* Universal Crossplane Version: crossplane:v1.11.2 
* Provider Version: xpkg.upbound.io/upbound/provider-aws:v0.31.0 
* Kubernetes version: v1.25.6

**Comment by fernandezcuesta** [comment]
One side-effect of this:

```
Message:               apply failed: creating Route Table (private-route-table-0-us-east-2-main) Association:  NotFound: The routeTable ID 'private-route-table-0-us-east-2-main' does not exist
status code: 400, request id: 67f46591-7f08-44df-94c4-d866c2aed3d7: 
```

**Comment by haarchri** [comment]
the external-name identifier for subnet is the id `subnet-...` from AWS 

https://github.com/upbound/provider-aws/blob/main/config/externalname.go#L280

if you want to set a name for subnet add a tag with Name key

**Comment by fernandezcuesta** [comment]
Hi, thanks for the heads-up, but that's not what I'm referring to.

Let me rephrase the issue in that case.
The purpose is not setting a the name of the object in AWS (as you can infer from the composition snippet, the Name tag  is already set with the same value).

The original goal was to allow Crossplane take ownership of infrastructure elements managed by a former setup of Crossplane. In other words, when Crossplane chart was reinstalled, multiple components were out of sync:

```
NAME                             READY   SYNCED   EXTERNAL-NAME                     AGE
private-route-0-us-east-2-main   True    False    rtb-0dde94591c6c2adc6_0.0.0.0/0   21h
private-route-1-us-east-2-main   True    False    rtb-04e63b3bda8e9710d_0.0.0.0/0   21h
private-route-2-us-east-2-main   True    False    rtb-0c628bbaf45ea0b5a_0.0.0.0/0   21h
````

and getting the following warnings from `provider-aws` deployment:

```
│ provider-aws-90b53987357c-85d764978-chj94 1.6790472021499872e+09    DEBUG    provider-aws    Cannot observe external resource    {"controller": "managed/ec2.aws.upbound.io/v1beta1, kind=route", "request":  │
│ "/private-route-1-us-east-2-main", "uid": "5b929868-785f-4477-86db-16b57465b577", "version": "2694401", "external-name": "rtb-04e63b3bda8e9710d_0.0.0.0/0", "error": "cannot run plan: plan failed: ", "error │
│ Verbose": "plan failed: \ncannot run plan\ngithub.com/upbound/upjet/pkg/controller.(*external).Observe\n\tgithub.com/upbound/upjet@v0.9.0-rc.0.0.20230308103251-5377e5db7994/pkg/controller/external.go:204\n │
│ github.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v0.19.0-rc.0.0.20221012013934-bce61005a175/pkg/reconciler/managed/reconci │
│ ler.go:780\ngithub.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile\n\tgithub.com/crossplane/crossplane-runtime@v0.19.0-rc.0.0.20221012013934-bce61005a175/pkg/ratelimiter/reconcile │
│ r.go:54\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:121\nsigs.k8s.io/controller-runtime/p │
│ kg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:320\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controlle │
│ r).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io │
│ /controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234\nruntime.goexit\n\truntime/asm_amd64.s:1594"}                                                                                           │
│ provider-aws-90b53987357c-85d764978-chj94 1.6790472021517222e+09    DEBUG    events    Warning    {"object": {"kind":"Route","name":"private-route-1-us-east-2-main","uid":"5b929868-785f-4477-86db-16b57465b │
│ 577","apiVersion":"ec2.aws.upbound.io/v1beta1","resourceVersion":"2694401"}, "reason": "CannotObserveExternalResource", "message": "cannot run plan: plan failed: "} 
```

The solution/workaround at that point was to explicitly set `external-name`, so Crossplane could regain ownership and SYNC these objects.
As a side effect, I got bouncing external-names as stated here and errors as raised [in this issue](https://github.com/upbound/provider-aws/issues/614).

**Comment by ONordander** [comment]
@fernandezcuesta did you ever find any workaround for this?

Also would be nice with some guidance from Upbound on how to handle resources with non-predictable external-names that are generated by AWS. Is there any other mechanism that we can use to inherit ownership of an already existing resource (without manually adding the annotation)?
